### PR TITLE
Correctly mapping overwrites when creating a channel and renamed all relevant property names as of #1562

### PR DESCRIPTION
**Pull request by SpaceEEC**
**Please describe the changes this PR makes and why it should be merged:**

Since [those](https://github.com/hydrabolt/discord.js/blob/1e47cfdd5df1654ac6ed47146e09f34e3e49b0c5/src/structures/PermissionOverwrites.js#L32-L33) two properties were renamed PermissionOverwrites require mapping before sending them to discord.

Changing the names of the properties back would be a better solution IMO.

Also passing now the overwrites as correct parameter when cloning a channel.

Edit:

Changed all `.deny` and `.allow` for permission related methods in `GuildChannel` to their new property names.
Related to #1595's problem with `.joinable` being incorrectly `true`.

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/Guild.js [hunk 1]
```diff
@@ -843,7 +843,14 @@ class Guild {
    *  .catch(console.error);
    */
   createChannel(name, type, { overwrites, reason } = {}) {
-    if (overwrites instanceof Collection) overwrites = overwrites.array();
+    if (overwrites instanceof Collection || overwrites instanceof Array) {
+      overwrites = overwrites.map(overwrite => ({
+        allow: overwrite.allow || overwrite._allowed,
```
Comment (PgBiel) [inline src/structures/Guild.js]: `.allowed`

File: src/structures/Guild.js [hunk 2]
```diff
@@ -843,7 +843,14 @@ class Guild {
...
+        deny: overwrite.deny || overwrite._denied,
```
Comment (PgBiel) [inline src/structures/Guild.js]: `.denied`

File: src/structures/GuildChannel.js [hunk 3]
```diff
@@ -76,20 +76,20 @@ class GuildChannel extends Channel {
     const overwrites = this.overwritesFor(member, true, roles);
 
     if (overwrites.everyone) {
-      permissions &= ~overwrites.everyone.deny;
-      permissions |= overwrites.everyone.allow;
+      permissions &= ~overwrites.everyone._denied;
+      permissions |= overwrites.everyone._alloweded;
```
Comment (bdistin) [inline src/structures/GuildChannel.js]: _alloweded => _allowed

Comment (SpaceEEC) [inline src/structures/GuildChannel.js]: nice copy paste :^)

**Pull request merged**